Update view-transitions.mdx (#9017)
withastro/docs: src/content/docs/en/guides/view-transitions.mdx

}
```

> The following example shows all the necessary properties to define a custom transition animation.

Reviewer: ```suggestion
```

> In this instance, the custom animation `bump` is defined inside a `<style is:global>` tag in your root layout file:

Reviewer: ```suggestion
The following example shows all the necessary properties to define a custom `bump` animation inside a `<style is:global>` tag in your root layout file:
```
Did we really need two sentences for this? Will this do?  :smile:

```astro title="src/layouts/Layout.astro
---
import { ViewTransitions } from 'astro:transitions';
---

<html lang="en">
  <head>
    <ViewTransitions />
  </head>
  <body>
    <slot />
  </body>
</html>

<style is:global>
  @keyframes bump {
    0% {
      opacity: 0;
      transform: scale(1) translateX(200px);
    }
    50% {
      opacity: 0.5;
      transform: scale(1.1);
    }
    100% {
      opacity: 1;
      transform: scale(1) translateX(0);
    }
  }
</style>

```

> Remember to define the animation behavior in the frontmatter of all the components using the animation:

Reviewer: ```suggestion
The animation behavior's must be defined in the frontmatter of every component using the animation:
```

```astro title="src/pages/index.astro"
---
const anim = {
  old: {
    name: 'bump',
    duration: '0.5s',
    easing: 'ease-in',
    direction: 'reverse,
  },
  new: {
    name: 'bump',
    duration: '0.5s',
    easing: 'ease-in-out',
  },
};

const customTransition = {
  forwards: anim,
  backwards: anim,
};
---

<header transition:animate={customTransition}> ... </header>
```

> When customizing animations, take some time to experiment with different variations. You can use different objects for forward and backward, and your keyframe animations for old and new can be different as well.

Reviewer: ```suggestion

You have great flexibility when defining custom animations. To achieve your desired result, you may wish to consider unusual combinations such as using different objects for forward and backward, or providing separate keyframe animations for old and new.
```

Is this sort of like what you were going for? It's not really docs' place to say "take some time to experiment", but I did try to make it sound like "you can probably do more than you think -- this feature isn't limiting"

}
```

The following example shows all the necessary properties to define a custom `bump` animation inside a `<style is:global>` tag in your root layout file:

```astro title="src/layouts/Layout.astro
---
import { ViewTransitions } from 'astro:transitions';
---

<html lang="en">
  <head>
    <ViewTransitions />
  </head>
  <body>
    <slot />
  </body>
</html>

<style is:global>
  @keyframes bump {
    0% {
      opacity: 0;
      transform: scale(1) translateX(200px);
    }
    50% {
      opacity: 0.5;
      transform: scale(1.1);
    }
    100% {
      opacity: 1;
      transform: scale(1) translateX(0);
    }
  }
</style>

```

> The animation behavior's must be defined in the frontmatter of every component using the animation:

Reviewer: ```suggestion
The animation's behavior must be defined in the frontmatter of every component using the animation:
```

}
```

The following example shows all the necessary properties to define a custom transition animation.

> In this instance, the custom animation `crash` is defined inside a `<style is:global>` tag in your root layout file:

Reviewer 2: "crash" sounds so destructive and somehow like "broken". Surely there's a more cheerful name?

```astro
---
import ViewTransitions from "astro/components/ViewTransitions.astro";
---

<html lang="en">
  <head>
    <meta charset="utf-8" />
    <link rel="icon" type="image/svg+xml" href="/favicon.svg" />
    <meta name="viewport" content="width=device-width" />
    <meta name="generator" content={Astro.generator} />
    <title>Astro</title>
    <ViewTransitions />
  </head>
  <body>
    <slot />
  </body>
</html>

<style is:global>
  @keyframes crash {
    0% {
      opacity: 0;
      transform: scale(1) translateX(200px);
    }
    50% {
      opacity: 0.5;
      transform: scale(1.1);
    }
    100% {
      opacity: 1;
      transform: scale(1) translateX(0);
    }
  }
</style>

```

Remember to define the animation behavior in the frontmatter of all the components using the animation:

```astro
---
const anim = {
  old: {

>     name: "crash",

Reviewer 2: Please use single quotation marks in the frontmatter, as is usually done in the documents

    duration: "0.5s",
    easing: "ease-in",

>     fillMode: "forwards",

Reviewer 2: ```suggestion
    direction: "reverse",
```

  },
  new: {
    name: "crash",
    duration: "0.5s",
    easing: "ease-in-out",

>     fillMode: "backwards",

Reviewer 2: ```suggestion
```

  },
};

const customTransition = {
  forwards: anim,
  backwards: anim,
};
---

<header transition:animate={customTransition}> ... </header>
```

> 

Reviewer 2: Maybe we should add a sentence saying that you can of course use different objects for forward and backward and different keframe names for old and new?
